File: src/firefly/java/edu/caltech/ipac/firefly/visualize/InsertBandInitializer.java
/*
 * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
 */
package edu.caltech.ipac.firefly.visualize;

import java.io.Serializable;
/**
 * User: roby
 * Date: Sep 11, 2009
 * Time: 11:16:02 AM
 */



/**
 * @author Trey Roby
 */
public class InsertBandInitializer implements Serializable {

    private final static String SPLIT_TOKEN= "--InsertBandInitializer--";

    private PlotImages    _initImages;
    private PlotState     _plotState;
    private WebFitsData   _fitsData;
    private Band          _band;
    private String        _dataDesc;


    public InsertBandInitializer(PlotState plotState,
                             PlotImages    images,
                             Band          band,
                             WebFitsData   fitsData,
                             String        dataDesc) {

        _plotState= plotState;
        _initImages= images;
        _band= band;
        _fitsData= fitsData;
        _dataDesc= dataDesc;
    }

//======================================================================
//----------------------- Public Methods -------------------------------
//======================================================================

    public PlotState getPlotState() { return _plotState; }
    public PlotImages getImages() { return _initImages; }
    public WebFitsData getFitsData()  { return _fitsData; }
    public Band getBand()  { return _band; }
    public String getDataDesc()  { return _dataDesc; }

    public String toString() {
        return _initImages +SPLIT_TOKEN+
               _plotState  +SPLIT_TOKEN+
               _fitsData   +SPLIT_TOKEN+
               _band       +SPLIT_TOKEN+
               _dataDesc;
    }


    private static String getString(String s) { return s.equals("null") ? null : s; }
}

File: src/firefly/js/visualize/InsertBandInitializer.js
/*
 * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
 */





import PlotState from './PlotState.js';
import PlotImages from './PlotImages.js';
import Band from './Band.js';
import WebFitsData from './WebFitsData.js';
import join from 'underscore.string/join';
import {parseInt, parseBoolean, parseFloat, checkNull,getStringWithNull } from '../util/StringUtils.js';

const SPLIT_TOKEN= "--InsertBandInitializer--";

class InsertBandInitializer {

    /**
     *
     * @param {PlotState} plotState
     * @param {PlotImages} images
     * @param {Band} band
     * @param {WebFitsData} fitsData
     * @param {string} dataDesc
     */
    constructor(plotState, images, band, fitsData, dataDesc) {
        this.plotState= plotState;
        this.initImages= images;
        this.band= band;
        this.fitsData= fitsData;
        this.dataDesc= dataDesc;
    }


//======================================================================
//----------------------- Public Methods -------------------------------
//======================================================================
    toString() {
        return join(SPLIT_TOKEN, this.initImages, this.plotState, this.fitsData, this.band, this.dataDesc);
    }

    static parse(s) {
        if (!s) return null;
        var sAry= s.split(SPLIT_TOKEN,6);
        if (sAry!==5) return null;

        var i= 0;
        var initImages= PlotImages.parse(sAry[i++]);
        var plotState= PlotState.parse(sAry[i++]);
        var fitsData= WebFitData.parse(sAry[i++]);
        var band= Band.parse(sAry[i++]);
        var dataDesc= getStringWithNull(sAry[i]);
        return new InsertBandInitializer(plotState,initImages,band,fitsData,dataDesc);
    }
}

export default InsertBandInitializer;
